From ce98ff71a879f2ee28817f231ae8930ce85d5c17 Mon Sep 17 00:00:00 2001
Message-Id: <ce98ff71a879f2ee28817f231ae8930ce85d5c17.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Tue, 10 Jun 2014 21:56:03 +0530
Subject: [PATCH 096/312] FOR_UPSTREAM [VPG]: drm/i915: Bit 29 of DPLLA or B
 needed for pallete registers

Bit 29 of DPLLA or B is required to be ON for using the pallete registers.
In case of MIPI, the DPLL will not be used but this bit is needed to
be ON. Else the system will hang.

Issue: GMIN-1472
Change-Id: Ie2cf3a9af96aecceba2ac97286eb4b59d21d36d2
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
Signed-off-by: Shashank Sharma <shashank.sharma@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi_pll.c |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dsi_pll.c b/drivers/gpu/drm/i915/intel_dsi_pll.c
index ba79ec1..3b20d81 100644
--- a/drivers/gpu/drm/i915/intel_dsi_pll.c
+++ b/drivers/gpu/drm/i915/intel_dsi_pll.c
@@ -268,6 +268,19 @@ void vlv_enable_dsi_pll(struct intel_encoder *encoder)
 	/* wait at least 0.5 us after ungating before enabling VCO */
 	usleep_range(1, 10);
 
+	I915_WRITE(_DPLL_A, I915_READ(_DPLL_A) | DPLL_REFA_CLK_ENABLE_VLV);
+	/*
+	 * Clock settle time. DSI PLL will be used
+	 * for DSI. But the palette registers
+	 * need REF clock of DPLLA or B to be
+	 * ON for functioning.
+	 * This settle time is required as DPLLA will be
+	 * unused earlier. Without this delay, system
+	 * goes to an unstable condition and throws
+	 * crash warnings.
+	 */
+	udelay(1000);
+
 	tmp = vlv_cck_read(dev_priv, CCK_REG_DSI_PLL_CONTROL);
 	tmp |= DSI_PLL_VCO_EN;
 	vlv_cck_write(dev_priv, CCK_REG_DSI_PLL_CONTROL, tmp);
-- 
1.7.9.5

